feat: Add option to control the length of Quake window animation

ArchieMeng opened the pull request:
Add a slider in advanced settings to control the duration of Quake window animation

zccrs commented on src/assets/other/default-config.json:
```diff
@@ -371,6 +371,14 @@
+                        },
+                        {
+                            "key": "quake_window_duration",
+                            "name": "Quake window duration",
```
这里是说窗口出来或隐藏时的动画时长吗？建议加上 Animation：`Quake window Animation duration`

ArchieMeng replied:
是的，指的是动画．

ArchieMeng commented:
/recheck

ArchieMeng commented:
/check build

zccrs commented on src/main/mainwindow.cpp:
```diff
@@ -3008,7 +3008,9 @@ int QuakeWindow::getQuakeAnimationTime()
     //动画时间计算方法：3quakeAnimationBaseTime加上(quakeAnimationHighDistributionTotalTime乘以当前雷神高度除以雷神最大高度)所得时间，为各个高度时动画时间
-    int durationTime = quakeAnimationBaseTime + quakeAnimationHighDistributionTotalTime * this->getQuakeHeight() / (screenRect.height() * 2 / 3);
+    int quakeAnimationBaseTime = Settings::instance()->QuakeDuration() * 2 / 3;
+    int quakeAnimationHighDistributionTotalTime = Settings::instance()->QuakeDuration() / 3;
```
这里为什么要除以3呢

ArchieMeng replied:
保持原来的quakeAnimationBaseTime和quakeAnimationHighDistributionTotalTime的时间比例．

zccrs commented on src/main/mainwindow.cpp:
```diff
@@ -3008,7 +3008,9 @@ int QuakeWindow::getQuakeAnimationTime()
     //quakeAnimationBaseTime+quakeAnimationHighDistributionTotalTime的时间是雷神窗口最大高度时动画效果时间
     //动画时间计算方法：3quakeAnimationBaseTime加上(quakeAnimationHighDistributionTotalTime乘以当前雷神高度除以雷神最大高度)所得时间，为各个高度时动画时间
-    int durationTime = quakeAnimationBaseTime + quakeAnimationHighDistributionTotalTime * this->getQuakeHeight() / (screenRect.height() * 2 / 3);
+    int quakeAnimationBaseTime = Settings::instance()->QuakeDuration() * 2 / 3;
```
这个值有什么用途？

ArchieMeng replied:
这个是不随雷神窗口高度变动的动画时长．

zccrs commented on src/main/service.cpp:
```diff
@@ -96,6 +96,7 @@ void Service::initSetting()
     // 关闭时delete
     m_settingDialog->widgetFactory()->registerWidget("fontcombobox", Settings::createFontComBoBoxHandle);
     m_settingDialog->widgetFactory()->registerWidget("slider", Settings::createCustomSliderHandle);
+    m_settingDialog->widgetFactory()->registerWidget("valslider", Settings::createValSliderHandle);
```
动画时间用slider不方便控制具体的值呀。是不是用 spinbox 会更适合些

ArchieMeng replied:
我的理解是动画时长是很感官上的，不需要精准控制具体的值，能方便地拉到合适的时长就够了．spinbox没法做到这样．

zccrs reviewed (changes requested):
只需要修改 deepin-terminal.ts 和  deepin-terminal_zh_CN.ts，其他语言不要在这里改动。

zccrs commented:
上个效果图看一下

ArchieMeng commented:
效果如视频所示：

https://user-images.githubusercontent.com/12139431/219543133-57dc487d-c608-4061-9177-61f3088f1845.mp4

ArchieMeng committed:
feat: Add option to control the length of Quake window animation

Add a slider in advanced settings to control the duration of Quake window animation

ArchieMeng commented:
https://user-images.githubusercontent.com/12139431/219989388-33306016-01d8-4ac1-8bb3-8beebb2e67d5.mp4

当前效果．

zccrs commented on src/settings/settings.cpp:
```diff
@@ -681,6 +689,40 @@ QPair<QWidget *, QWidget *> Settings::createCustomSliderHandle(QObject *obj)
+    {
+        valTicksList << "";
+    }
+    valTicksList << tr("Long");
```
建议用“快”和“慢”来表示

ArchieMeng replied:
修改好了．
